File: 1148-Article-Views-I-EASY/soln.py
import pandas as pd

def self_views(views: pd.DataFrame) -> pd.DataFrame:
    mask = views["author_id"] == views["viewer_id"]
    cols = ["author_id"]
    
    result = views.loc[mask, cols]
    result = result.drop_duplicates().rename(columns={"author_id": "id"})
    
    return result.sort_values("id").reset_index(drop=True)

File: 1148-Article-Views-I-EASY/soln.scala
import org.apache.spark.sql.DataFrame
import org.apache.spark.sql.functions.col

object Solution {
  def selfViews(views: DataFrame): DataFrame = {
    views
      .filter(col("author_id") === col("viewer_id"))
      .select(col("author_id").alias("id"))
      .distinct()
      .orderBy("id")
  }
}
